identity/aziot-identityd/src/lib.rs

```diff
@@ -702,76 +634,199 @@ impl Api {
         Ok(device)
     }
 
+    async fn issue_local_identity(
+        &self,
+        module_id: &str,
+    ) -> Result<aziot_identity_common::Identity, Error> {
+        let localid = self.settings.localid.as_ref().ok_or_else(|| {
+            Error::Internal(InternalError::BadSettings(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "no local id settings specified",
+            )))
+        })?;
+
+        let local_identity =
+            match self
+                .local_identities
+                .get(&aziot_identity_common::ModuleId(module_id.to_owned()))
+            {
+                None => {
+                    return Err(Error::invalid_parameter(
+                        "module_id",
+                        format!("no local identity found for {}", module_id),
+                    ))
+                }
+                Some(opts) => {
+                    let attributes = opts.as_ref().map_or(
+                        aziot_identity_common::LocalIdAttr::default(),
+                        |opts| match opts {
+                            settings::LocalIdOpts::X509 { attributes } => *attributes,
+                        },
+                    );
+
+                    let subject = format!("{}.{}", module_id, localid.domain);
+                    let (certificate, private_key) = self
+                        .create_identity_cert_if_not_exist_or_expired(
+                            KeyType::Raw,
+                            module_id,
+                            subject.as_str(),
+                            Some(attributes),
+                        )
+                        .await?;
+
+                    // Parse certificate expiration time.
+                    let expiration = get_cert_expiration(&certificate)?;
+
+                    aziot_identity_common::Identity::Local(aziot_identity_common::LocalIdSpec {
+                        private_key,
```

Comment:
We should add the module_id in the response.

Reply:
Added